markdown/report.Rmd

```diff
@@ -4,8 +4,82 @@ output:
+source(here::here("functions", "summarise_data.R"))
+source(here::here("functions", "calculate_qom.R"))
+source(here::here("functions", "day_diff.R"))
+
+# Remove tidylog
+detach("package:tidylog", unload = TRUE)
+
+# Load basefile
+basefile <- read_rds(here::here("data", "basefiles", 
+                                glue("{pub_date}_base-file.rds")))
+
+# Basefile for most recent FY
+latest <- summarise_data(basefile)
+
+# Health Board data
+hb <- basefile %>% summarise_data(hb)
+max_hb <- hb %>% filter(qom == max(qom))
+min_hb <- hb %>% filter(qom == min(qom))
+
+# HSCP data
+hscp <- basefile %>% summarise_data(hscp)
+max_hscp <- hscp %>% filter(qom == max(qom))
+min_hscp <- hscp %>% filter(qom == min(qom))
+
+# Age Gender data
+allages <- basefile %>% summarise_data(sex)
+old     <- basefile %>% filter(age_grp == "85+" & !is.na(sex)) %>% summarise_data(sex)
+young   <- basefile %>% filter(age_grp == "0-54" & !is.na(sex)) %>% summarise_data(sex)
+
+max_old   <- old %>% filter(qom == max(qom)) %>% pull(sex) %>% tolower()
+min_old   <- old %>% filter(qom == min(qom)) %>% pull(sex) %>% tolower()
+max_young <- young %>% filter(qom == max(qom)) %>% pull(sex) %>% tolower()
+min_young <- young %>% filter(qom == min(qom)) %>% pull(sex) %>% tolower()
+
+# SIMD data
+simd <- basefile %>% summarise_data(simd)
+max_simd <- simd %>% filter(qom == max(qom)) %>% pull(simd)
+min_simd <- simd %>% filter(qom == min(qom)) %>% pull(simd)
+
+# Urban Rural data
+ur <- basefile %>% summarise_data(urban_rural) %>%
+  mutate(urban_rural = 
+           case_when(
+             str_detect(urban_rural, "Rural$") ~ paste(tolower(substring(urban_rural, 3)), "areas"),
+             TRUE ~ tolower(substring(urban_rural, 3))
+           ))
+max_ur <- ur %>% filter(qom == max(qom)) %>% pull(urban_rural)
+min_ur <- ur %>% filter(qom == min(qom)) %>% pull(urban_rural)
 
-```{r ListofTables, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE}
+```
+
+```{r links, echo = FALSE, message = FALSE, warning = FALSE, error = FALSE}
+
+data_tables <-
+  glue("http://www.isdscotland.org/Health-Topics/Health-and-Social-Community-Care/Publications/",
+       "{pub_date}/QOM_tables_{year(pub_date)}{format(pub_date, '%m')}.xls")
+
+```
+
+```{r contact-table, echo=FALSE, message=FALSE, warning=FALSE,error=FALSE}
+# Create contact information
+contact1 <- c("**Federico Centoni**", "Information Analyst", 
+              "0131 314 1532", "federico.centoni@nhs.net")
+contact2 <- c("**Alice Byers**", "Senior Information Analyst", 
+              "0131 314 1725", "alice.byers@nhs.net")
+contact3 <- c("**Aghimien Iyayi-Igbinovia**", "Principal Information Analyst", 
+              "0141 207 1616", "aghimien.iyayi-igbinovia@nhs.net")
+
+contact <- data.frame(contact1, contact2, contact3)
+```
+
+```{r list-of-tables, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE}
 ## Automated method for adding file names and sizes
 # get list of files for the publication
 file_list <- list.files() 
```

Comment:
I reckon we may need to create a folder called "datafiles" under "data" folder (where basefiles and extracts folders are) to store the excel workbooks. If that's the case, we need to put the file path in list.files(), and for paths[i]:

```
file_list <- list.files(file.path(here(), "data/datafile"))
```

```
path[i] <- paste0(file.path(here(), "data/datafile"), "/", file_list[i])
```

Reply:
Thanks Tina - I've created an 'output' folder and updated this section. I think this bit of code could be improved on - I know hsmr have a file size function we might be able to use. More of a 'nice to have' though so will look into this later.